### DISCUSSION TITLE
Alpha in Color renders differently than standard?

### DISCUSSION THREAD

**Discussion by brinko99** [dotnet/wpf#7710] [Q&A]
I’m trying to implement a text color in WPF that matches CSS RGBA of `#00,00,00,0.72` on a white background. I’d expect that to be a color of `#B8000000` (alpha of dec 184 ≈ .72), yet the result is much lighter than expected. Applying opacity to the control itself works as expected.

I'm expecting the composite color to be `#474747` as produced by CSS and paint.net, but I'm getting `#8C8C8C`. Does WPF render alpha on a brush differently somehow?

Here is an example:
![image](https://user-images.githubusercontent.com/1231765/231280155-d32f2c70-b563-4bc7-b1aa-e10208978ae8.png)

```
<StackPanel Background="White">
    <TextBlock Text="Medium Contrast (#B8000000)" Foreground="#B8000000" Style="{x:Null}" />
    <TextBlock Text="Medium Contrast (Opacity .72)" Foreground="#FF000000" Opacity=".72"  Style="{x:Null}"/>
</StackPanel>
```

Any help is appreciated!

**Comment by miloush** [comment]
Hmm, good catch. This seems to be an issue only for text and only for certain font sizes (small enough). `TextOptions.TextFormattingMode="Display"` seems to affect it, but does not always fix it. It looks like a bug to me.

```xaml
<StackPanel>
    <Glyphs FontUri="C:\WINDOWS\Fonts\TIMES.TTF" FontRenderingEmSize="33" UnicodeString="Hello World!" Fill="#B8000000" />
    <Glyphs FontUri="C:\WINDOWS\Fonts\TIMES.TTF" FontRenderingEmSize="34" UnicodeString="Hello World!" Fill="#B8000000" />
</StackPanel>
```

<img width="277" alt="image" src="https://user-images.githubusercontent.com/10546952/231294098-ec91c0bb-585c-4965-8db0-26fa4d75476f.png">


Same with `<GlyphRunDrawing>` but it depends on its size, not `FontRenderingEmSize`.

**Reply by brinko99** [reply]
Thanks for the confirmation... I'll create an issue for this.

**Reply by lindexi** [reply]
@miloush I think it might have something to do with how GlyphRun is rendered. On the GFX module of WPF, it will decide whether to use Geometry to draw or DrawGlyphs to draw according to the drawing size of the text. But I don't have a deeper understanding of the problem. And I wrote the [blog](https://blog-lindexi-com.translate.goog/post/dotnet-%E8%AF%BB-WPF-%E6%BA%90%E4%BB%A3%E7%A0%81%E7%AC%94%E8%AE%B0-%E6%B8%B2%E6%9F%93%E5%B1%82%E6%98%AF%E5%A6%82%E4%BD%95%E5%B0%86%E5%AD%97%E7%AC%A6-GlyphRun-%E7%94%BB%E5%87%BA%E6%9D%A5%E7%9A%84.html?_x_tr_sl=zh-CN&_x_tr_tl=en&_x_tr_hl=zh-CN&_x_tr_pto=wapp) about it.
